packages/cpp/ArmoniK.Api.Worker/header/utils/WorkerServer.h

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file WorkerServer.h
+ * @brief Contains the WorkerServer class, which represents the worker server
+ * for ArmoniK API.
+ */
+#pragma once
+#include <grpcpp/create_channel.h>
+#include <grpcpp/security/credentials.h>
+
+#include <functional>
+#include <memory>
+
+#include "agent_common.pb.h"
+#include "agent_service.grpc.pb.h"
+#include "grpcpp/server_builder.h"
+#include "options/ComputePlane.h"
+#include "serilog/SerilogContext.h"
+#include "serilog/serilog.h"
+#include "utils/IConfiguration.h"
+
+using namespace armonik::api::grpc::v1::agent;
+
+namespace armonik::api::worker {
+/**
+ * @class WorkerServer
+ * @brief Represents the worker server for ArmoniK API.
+ */
+class WorkerServer {
+ public:
+  common::serilog::serilog logger;
+
+ private:
+  ::grpc::ServerBuilder builder_;
+  std::shared_ptr<common::utils::IConfiguration> configuration_;
```

Comment:
I would say it is not necessary to store the configuration and the builder during the lifetime of the server.